controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -145,15 +145,24 @@ private void triggerReadyJobs(LockedApplication application) {
         List<JobType> jobs =  order.jobsFrom(application.deploymentSpec());
 
         // Should the first step be triggered?
-        // TODO: How can the first job not be systemTest (second ccondition)?
-        if ( ! jobs.isEmpty() && jobs.get(0).equals(JobType.systemTest) &&
-             application.deploying().get() instanceof Change.VersionChange) {
-            Version target = ((Change.VersionChange)application.deploying().get()).version();
-            JobStatus jobStatus = application.deploymentJobs().jobStatus().get(JobType.systemTest);
-            if (jobStatus == null || ! jobStatus.lastTriggered().isPresent() 
-                || ! jobStatus.lastTriggered().get().version().equals(target)) {
-                application = trigger(JobType.systemTest, application, false, "Upgrade to " + target);
-                controller.applications().store(application);
+        if ( ! jobs.isEmpty() && jobs.get(0).equals(JobType.systemTest) ) {
```

Comment:
How can the first job not be system test, though? Isn't the deplyment spec made such that it always is, unless it isEmpty()?

Reply:
Yes, but that code is far away and not explicitly guaranteed in the API, so I don't feel comfortable implicitly assuming it.

Reply:
Fair enough.